### PR TITLE
documentation setup

### PR DESCRIPTION
**Pull request by saurabhintoml**

### CODE REVIEW

**Review by dhendryc — CHANGES_REQUESTED** [review]

File: Manifest.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# This file is machine-generated - editing it directly is not advised
```
Comment (matbesancon) [inline Manifest.toml:1]: the manifest file should not be committed

File: LICENSE [hunk 2]
```diff
@@ -1,5 +1,6 @@
 MIT License
 
+Copyright (c) 2024 Saurabh Srivastava
```
Comment (matbesancon) [inline LICENSE:3]: You don't need to add individual contributors to the license file, they are kept in the git tracker instead

File: README.md [hunk 3]
```diff
@@ -7,39 +7,55 @@ A solver for Mixed-Integer Convex Optimization that uses Frank-Wolfe methods for
 
 ## Overview
 
-The Boscia.jl solver combines (a variant of) the Frank-Wolfe algorithm with a branch-and-bound like algorithm to solve mixed-integer convex optimization problems of the form
-`min_{x ∈ C, x_I ∈ Z^n} f(x)`,
-where `f` is a differentiable convex function, `C` is a convex and compact set, and `I` is a set of indices of integral variables.
+The Boscia.jl solver uses a combination of a variant of the Frank-Wolfe algorithm and a branch-and-bound-like algorithm to solve mixed-integer convex optimization problems. These problems are of the form:
```
Comment (matbesancon) [inline README.md:10]: the sentence here is too verbose

File: README.md [hunk 4]
```diff
@@ -7,39 +7,55 @@ A solver for Mixed-Integer Convex Optimization that uses Frank-Wolfe methods for
...
+**min_{x ∈ C, x_I ∈ Z^n} f(x)**,
```
Comment (matbesancon) [inline README.md:11]: please revert the changes here on line 11 and 12, they are not necessary

File: README.md [hunk 5]
```diff
@@ -7,39 +7,55 @@ A solver for Mixed-Integer Convex Optimization that uses Frank-Wolfe methods for
...
 
 `Boscia.jl` uses [`FrankWolfe.jl`](https://github.com/ZIB-IOL/FrankWolfe.jl) for solving the convex subproblems, [`Bonobo.jl`](https://github.com/Wikunia/Bonobo.jl) for managing the search tree, and oracles optimizing linear functions over the feasible set, for instance calling [SCIP](https://scipopt.org) or any MOI-compatible solver to solve MIP subproblems.
 
 ## Installation
 
+
+Once you have installed Julia , From the Julia REPL, type ] to enter the Pkg REPL mode and run 
+```Boscia
+pkg > add Boscia
+
```
Comment (matbesancon) [inline README.md:27]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -7,39 +7,55 @@ A solver for Mixed-Integer Convex Optimization that uses Frank-Wolfe methods for
...
+```
+or alternatively you can do this
```
Comment (matbesancon) [inline README.md:29]:
```suggestion
or alternatively via Pkg in any Julia code:
```

File: README.md [hunk 7]
```diff
@@ -7,39 +7,55 @@ A solver for Mixed-Integer Convex Optimization that uses Frank-Wolfe methods for
...
 ```
 
-Or get the latest master branch with:
-```julia
-import Pkg
-Pkg.add(url="https://github.com/ZIB-IOL/Boscia.jl", rev="main")
-```
+
+
+If you don't have SCIP  , you can on go this link and add SCIP as instructed ['SCIP'](https://github.com/scipopt/SCIP.jl)
```
Comment (matbesancon) [inline README.md:40]:
```suggestion
If you don't have SCIP, you can on go this link and add SCIP as instructed ['SCIP'](https://github.com/scipopt/SCIP.jl)
```

File: README.md [hunk 8]
```diff
@@ -7,39 +7,55 @@ A solver for Mixed-Integer Convex Optimization that uses Frank-Wolfe methods for
...
+If you want to use SCIP within Boscia and your OS is windows, you will have download SCIP separately, see SCIP.jl.
+Note that you do not necessarily have to download the binaries but can also use the installer provided by SCIP.
+
+
+**For Window Users** You need not to download whole SCIP binary instead you can follow **Custom Installation** mentioned on this page and download and link SCIP with your JULIA .
```
Comment (matbesancon) [inline README.md:45]: this is a bit too verbose, this is repeating the SCIP.jl readme

File: README.md [hunk 9]
```diff
@@ -7,39 +7,55 @@ A solver for Mixed-Integer Convex Optimization that uses Frank-Wolfe methods for
...
+
+
 
 ## Getting started
 
 Here is a simple example to get started. For more examples, see the examples folder in the package.
 
+
 ```julia
+
```
Comment (matbesancon) [inline README.md:58]:
```suggestion
```
empty line not necessary

File: examples/HiGHS_example.jl [hunk 10]
```diff
@@ -1,4 +1,11 @@
-using Boscia
+# This script sets up and solves a linear optimization problem using the Boscia and FrankWolfe packages in Julia. 
+# The problem involves a 6-dimensional variable `x` that is constrained to be between 0 and 1 (inclusive) and is also restricted to binary values (0 or 1).
+# The objective function `f(x)` is a quadratic function representing the squared Euclidean distance from a vector `diffw` (which is set to 0.5 for all components).
+# The gradient of the objective function is computed by `grad!`.
+# The `HiGHS.Optimizer` is used to handle the optimization problem and the `FrankWolfe.MathOptLMO` interface connects it to the Frank-Wolfe algorithm.
+# The final solution `x` and the result status are obtained using `Boscia.solve`.using Boscia
```
Comment (matbesancon) [inline examples/HiGHS_example.jl:6]:
```suggestion
# The final solution `x` and the result status are obtained using `Boscia.solve`.using Boscia
```

the `using Boscia` part should be on a new line

File: examples/approx_planted_point.jl [hunk 11]
```diff
@@ -1,3 +1,12 @@
+
+# This script tests different methods for solving optimization problems involving approximate planted points
```
Comment (matbesancon) [inline examples/approx_planted_point.jl:2]: I really like the addition of an introductory comment on the top of each example, but we should be a bit more concise for each of them. Anything that can be read from the code is not necessary. For instance 
> checks the solution against expected values, and verifies the correctness of  the results using `@test` assertions.

File: docs/src/basics.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
+# How does it work  ?
+
+Boscia focuses on mixed-integer convex problems in which the nonlinear constraints and objectives are convex and presents a new algorithmic framework for solving these problems that exploit recent advances in so-called Frank-Wolfe (FW) or Conditional Gradient (CG) methods. The problem class we consider is of the type:
```
Comment (dhendryc) [inline docs/src/basics.md:3]: Currently we only support linear constraints.

File: docs/src/basics.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+where \( X \) is a compact nonconvex set admitting a boundable linear minimization oracle (LMO), i.e., a set over which optimizing a linear function can be done efficiently (comparatively to the original problem), even when bounds are added or modified. Formally, we consider we have access to an oracle taking new bounds \( (l, u) \) and a direction \( d \):
+
+Given : 
+$$
+\( (l, u, d) \in \mathbb{R}^n \times \mathbb{R}^n \times \mathbb{R}^n \),  \( \arg\min_{v \in \mathbb{R}^n} \langle v, d \rangle \) 
+ \text {subject to}    \( v \in X \cap [l, u] \).
+$$
+
+
+## BPCG
```
Comment (dhendryc) [inline docs/src/basics.md:17]: BPCG is the default but not the only variant that can be used. I would move this more back.

File: docs/src/basics.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+
+## FW gap based termination 
+
+Frank-Wolfe methods produce primal feasible iterates and an FW gap, offering many inexpensive iterations with a gradually increasing dual bound. This allows early termination of nodes when the dual bound reaches the best incumbent's objective value, avoiding unnecessary computations. Nodes can be stopped anytime to produce a useful dual bound, aiding overall progress. This flexibility contrasts with other nonlinear solvers, enabling more efficient optimization.
+
+## Tree stae-dependent termination and evolving error 
+
+We implement termination criteria in node processing to reduce iterations, prioritizing nodes with promising lower bounds. An adaptive Frank-Wolfe gap criterion increases precision with depth in the branch-and-bound tree. BPCG's convex combinations of integer extreme points ensure rapid convergence, even with reduced precision runs. This approach balances efficiency and accuracy in solving optimization problems.
+
+## Strong Branching
```
Comment (dhendryc) [inline docs/src/basics.md:35]: I would add that we have Hybrid branching which couples strong branching and most infeasible branching. Strong branching is very expensive to do for the whole tree and it's usually more beneficial to utilize strong branching only up to a certain depth.

File: docs/src/index.md [hunk 15]
```diff
@@ -0,0 +1,141 @@
+# Boscia.jl
```
Comment (dhendryc) [inline docs/src/index.md:1]: We already have that in the `ReadMe` file.

File: docs/src/reference/1_algorithms.md [hunk 16]
```diff
@@ -0,0 +1,12 @@
+# Algorithms
```
Comment (dhendryc) [inline docs/src/reference/1_algorithms.md:1]: Algorithm 

Not Algorithms. Boscia is one algorithm with many options and settings.

File: docs/src/reference/1_algorithms.md [hunk 17]
```diff
@@ -0,0 +1,12 @@
+# Algorithms 
+
+This section contains information about interface.jl
```
Comment (dhendryc) [inline docs/src/reference/1_algorithms.md:3]: In this section we present the algorithm interface and the possible settings.

File: docs/src/reference/2_blmo_build.md [hunk 18]
```diff
@@ -0,0 +1,24 @@
+# LMO
```
Comment (dhendryc) [inline docs/src/reference/2_blmo_build.md:1]: The Bounded Linear Minimization Oracle (BLMO)

File: docs/src/reference/2_blmo_build.md [hunk 19]
```diff
@@ -0,0 +1,24 @@
+# LMO 
+
+This section contains information about interface blmo_interface.jl , build_lmo.jl and time_tracking_lmo.jl
```
Comment (dhendryc) [inline docs/src/reference/2_blmo_build.md:3]: This section describes the general interface to the Bounded Linear Minimization Oracle and the preimplemented types of BLMO.

File: docs/src/reference/2_blmo_build.md [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+
+## BLMO_Interface
```
Comment (dhendryc) [inline docs/src/reference/2_blmo_build.md:5]: General BLMO interface

File: docs/src/reference/2_blmo_build.md [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+```
+
+## Build LMO
+
+```@autodocs
+Modules = [Boscia]
+Pages = ["build_lmo.jl"]
+```
+
+## Time Tracking LMO
```
Comment (dhendryc) [inline docs/src/reference/2_blmo_build.md:19]: Time tracking wrapper

File: docs/src/reference/2_blmo_build.md [hunk 22]
```diff
@@ -0,0 +1,24 @@
...
+This section contains information about interface blmo_interface.jl , build_lmo.jl and time_tracking_lmo.jl
+
+## BLMO_Interface 
+
+```@autodocs
+Modules = [Boscia]
+Pages = ["blmo_interface.jl"]
+```
+
+## Build LMO
```
Comment (dhendryc) [inline docs/src/reference/2_blmo_build.md:12]: Adapting the BLMO to the node problems

Also move that behind the Time Tracking LMO

File: docs/src/reference/2_blmo_build.md [hunk 23]
```diff
@@ -0,0 +1,24 @@
...
+Modules = [Boscia]
+Pages = ["build_lmo.jl"]
+```
+
+## Time Tracking LMO
+
+```@autodocs
+Modules = [Boscia]
+Pages = ["time_tracking_lmo.jl"]
+```
```
Comment (dhendryc) [inline docs/src/reference/2_blmo_build.md:24]: Also, it's missing the `MOI oracle`, the `Managed_BLMO` and the `Polytopes`.

File: docs/src/reference/custom.md [hunk 24]
```diff
@@ -0,0 +1,24 @@
+# CusBono IntBound Problem
```
Comment (dhendryc) [inline docs/src/reference/custom.md:1]: The branch-and-bound tree functionality

File: docs/src/reference/custom.md [hunk 25]
```diff
@@ -0,0 +1,24 @@
+# CusBono IntBound Problem 
+
+This section contain information about custom_bonobo.jl , integer_bounds.jl and problem.jl
```
Comment (dhendryc) [inline docs/src/reference/custom.md:3]: This section contains the customization of the branch-and-bound implementation of Bonobo.jl.

We don't need the subsection here.

File: docs/src/reference/fw_variant.md [hunk 26]
```diff
@@ -0,0 +1,10 @@
+# FW_variant
```
Comment (dhendryc) [inline docs/src/reference/fw_variant.md:1]: The Frank-Wolfe variants.

File: docs/src/reference/fw_variant.md [hunk 27]
```diff
@@ -0,0 +1,10 @@
...
+
+This file contains information about all the frank_wolfe_variants.jl
```
Comment (dhendryc) [inline docs/src/reference/fw_variant.md:3]: Here is the list with the currently supported Frank-Wolfe variants.

File: docs/src/reference/utilities.md [hunk 28]
```diff
@@ -0,0 +1,10 @@
+# Utilities 
+
+This section contain information about utilities.jl
```
Comment (dhendryc) [inline docs/src/reference/utilities.md:3]: This section contains some utility functionality.

File: docs/src/reference/utilities.md [hunk 29]
```diff
@@ -0,0 +1,10 @@
...
+
+## utilities
```
Comment (dhendryc) [inline docs/src/reference/utilities.md:5]: Not necessary

**Review by dhendryc — CHANGES_REQUESTED** [review]

File: docs/src/basics.md [hunk 30]
```diff
@@ -0,0 +1,50 @@
+# How does it work  ?
+
+Boscia focuses on mixed-integer convex problems in which the linear constraints and objectives are convex and presents a new algorithmic framework for solving these problems that exploit recent advances in so-called Frank-Wolfe (FW) or Conditional Gradient (CG) methods. The problem class we consider is of the type:
```
Comment (dhendryc) [inline docs/src/basics.md:3]: constraints linear, objective nonlinear

**Review by dhendryc — CHANGES_REQUESTED** [review]

File: docs/src/basics.md [hunk 31]
```diff
@@ -0,0 +1,47 @@
...
+The classic FW algorithm's reliance on an LMO is inefficient for polytopes due to expensive calls. The Blended Pairwise Conditional Gradient (BPCG) algorithm improves efficiency by combining FW steps with pairwise steps that avoid LMO calls. BPCG maintains a smaller active set of vertices, enhancing performance. We use a modified lazified BPCG from FrankWolfe.jl.
+
+# Branch and Bound Techniques 
+
+In this section, we present the techniques derived from branch-and-bound techniques that can be used in our
+framework .
+
+## FW gap based termination 
+
+Frank-Wolfe methods produce primal feasible iterates and an FW gap, offering many inexpensive iterations with a gradually increasing dual bound. This allows early termination of nodes when the dual bound reaches the best incumbent's objective value, avoiding unnecessary computations. Nodes can be stopped anytime to produce a useful dual bound, aiding overall progress. This flexibility contrasts with other nonlinear solvers, enabling more efficient optimization.
```
[the remark this answers is not in the document]
Comment (saurabhintoml) [reply docs/src/basics.md:29]: okay

File: docs/src/basics.md (hunk 13 above)
Comment (saurabhintoml) [reply docs/src/basics.md:17]: what to do in this

File: examples/cube_blmo.jl [hunk 32]
```diff
@@ -1,4 +1,32 @@
 ## How to implement the BLMO Interface using the cube as an example
+# The code provided is an implementation of a Bounded Linear Minimization Oracle (BLMO) for a cube, using the Julia programming language and leveraging the Boscia and Bonobo libraries. This BLMO is designed to work within the structure of these libraries to facilitate linear minimization problems within bounded constraints.
+#
+# Key Components and Functions:
+#
+# 1. CubeBLMO Structure:
+#    - `CubeBLMO` is a mutable struct representing the BLMO with fields for the number of dimensions (`n`), integer variables (`int_vars`), bounds (`bounds`), and the time taken for solving (`solving_time`).
+#
+# 2. Necessary Functions:
```
[the remark this answers is not in the document]
Comment (saurabhintoml) [reply examples/cube_blmo.jl:9]: okay

File: .github/workflows/CI.yml [hunk 33]
```diff
@@ -0,0 +1,79 @@
+name: CI
```
Comment (matbesancon) [inline .github/workflows/CI.yml:1]: you should merge with ci.yaml (small)

File: LICENSE (hunk 2 above)
Comment (matbesancon) [reply LICENSE:3]: @saurabhintoml still to remove?

File: .github/workflows/CI.yml
Comment (dhendryc) [inline .github/workflows/CI.yml:1]: We don't need this file. Please merge it contents with `ci.yml`.

File: Manifest.toml
Comment (dhendryc) [inline Manifest.toml:1]: This should not be tracked. Please remove it from the git tracking list.

File: README.md [hunk 34]
```diff
@@ -7,38 +7,52 @@ A solver for Mixed-Integer Convex Optimization that uses Frank-Wolfe methods for
...
 ```julia
 import Pkg
-Pkg.add(url="https://github.com/ZIB-IOL/Boscia.jl", rev="main")
+Pkg.add("Boscia")
 ```
 
+
+
+Suggested change
+If you don't have SCIP  , you can on go this link and add SCIP as instructed ['SCIP'](https://github.com/scipopt/SCIP.jl)
```
Comment (dhendryc) [inline README.md:37]: We don't have to explicitly state this.

File: docs/make.jl [hunk 35]
```diff
@@ -0,0 +1,51 @@
...
+
+# Call the functions to update index.md and 1_algorithms.md
+copy_readme_to_index()
+
+
+# Generate documentation
+makedocs(
+    sitename = "Boscia.jl",
+    modules = [Boscia],
+    format = Documenter.HTML(repolink = "https://github.com/saurabhintoml/Boscia.jl.git"),
```
Comment (dhendryc) [inline docs/make.jl:24]: The links have to be changed to the main Boscia repository, i.e. the ZIB-IOL/Boscia one.

File: docs/src/basics.md (hunk 13 above)
Comment (dhendryc) [reply docs/src/basics.md:17]: Move it further down the page and state that it is used as the default.

File: docs/src/basics.md (hunk 31 above)
[the remark this answers is not in the document]
Comment (dhendryc) [reply docs/src/basics.md:29]: Where is the explanation of the Frank-Wolfe gap?

File: examples/mps-example.jl [hunk 36]
```diff
@@ -1,4 +1,10 @@
-using Boscia
```
Comment (dhendryc) [inline examples/mps-example.jl:1]: We still need Boscia as a dependency

File: examples/low_dim_in_high_dim.jl [hunk 37]
```diff
@@ -1,4 +1,10 @@
-using Boscia
```
Comment (dhendryc) [inline examples/low_dim_in_high_dim.jl:1]: We still need Boscia as a dependency.

File: examples/lasso.jl [hunk 38]
```diff
@@ -1,4 +1,12 @@
-using Statistics
```
Comment (dhendryc) [inline examples/lasso.jl:1]: The example requires this package.

**Review by dhendryc — CHANGES_REQUESTED** [review]
We don't need this file. Please merge it with the `ci.yml` file (in terms of contents).

**Pull request merged**